app/uk/gov/hmrc/apisubscriptionfields/repository/MongoCrudHelper.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apisubscriptionfields.repository
+
+import play.api.Logger
+import play.api.libs.json.{JsObject, OWrites, Reads}
+import reactivemongo.api.{Cursor, ReadPreference}
+import reactivemongo.play.json.ImplicitBSONHandlers._
+import reactivemongo.play.json.collection.JSONCollection
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+
+
+trait MongoCrudHelper[T] extends MongoIndexCreator with MongoErrorHandler {
+
+  val col: JSONCollection
+
+  def save(entity: T, selector: JsObject)(implicit w: OWrites[T]): Future[Boolean] = {
+    Logger.debug(s"[save] entity: $entity selector: $selector")
+    col.update(selector = selector, update = entity, upsert = true).map {
+      updateWriteResult => handleSaveError(updateWriteResult, s"Could not save entity: $entity")
+    }
+  }
+
+  def getCursor(selector: JsObject)(implicit r: Reads[T]):Future[List[T]] = {
+    Logger.debug(s"[getCursor] selector: $selector")
```

Comment:
Entity is part of the log statement in save() - consider adding it here. Not essential as entity could be inferred from selector but it would be clearer.